src/LiveComponent/tests/Integration/LiveComponentHydratorTest.php

```diff
@@ -783,6 +785,109 @@ public function provideDehydrationHydrationTests(): iterable
             ;
         }, 80100];
 
+        yield 'Object: (de)hydrates DTO correctly' => [function () {
+            return HydrationTest::create(new class() {
+                #[LiveProp(writable: true)]
+                public ?Address $address = null;
+
+                public function mount()
+                {
+                    $this->address = new Address();
+                    $this->address->address = '1 rue du Bac';
+                    $this->address->city = 'Paris';
+                }
+            })
+                ->mountWith([])
+                ->assertDehydratesTo([
+                    'address' => [
+                        'address' => '1 rue du Bac',
+                        'city' => 'Paris',
+                    ],
+                    '@checksum' => '2P/KES0BbGw1KNZsOn4CSEpN1HJ1s...+bQPY=',
```

Comment:
Not needed: the code that runs the test adds the correct `@checksum` for you. If you remove this, test still passes :)